src/Database/Connection/Adapter/MysqlAdapter.php

```diff
@@ -48,15 +48,18 @@ public function connection()
             $hostname = $this->config['hostname'];
         }
 
-        $port = '';
+        $port = ';port=';
 
         if ($hostname !== 'localhost' || $hostname == '127.0.0.1') {
             if (isset($this->config['port']) && !is_null($this->config['port']) && !empty($this->config['port'])) {
-                $port = ':'.$this->config['port'];
+                $port .= $this->config['port'];
             } else {
-                $port = ':'.self::PORT;
+                $port .= self::PORT;
             }
         }
+        if ($port === ';port=') {
+            $port .= '3306';
+        }
```

Comment:
Fixing the PHP PDO DSN generating issue when concatenating the database connection string with port number.